src/App.js

```diff
@@ -3,101 +3,41 @@ import './App.css';
 
 import Chess from "chess.js";
 import Chessboard from "chessboardjsx";
-import ChessWeight from "./ChessWeight";
+import { bestMove } from "./ai"
 
 const minimaxDepth = 3;
 
 export default function App() {
     const [game, setGame] = useState(new Chess());
     const [fen, setFen] = useState("start");
 
-    const minimaxRoot = (depth, wantMax) => {
-        const moves = game.moves();
-        if (game.game_over() || game.in_draw() || moves.length === 0) return;
-
-        let best = -9999;
-        let bestMove;
-        for (let i = 0; i < moves.length; ++i) {
-            game.move(moves[i]);
-            const value = minimax(depth - 1, -10000, 10000, !wantMax);
-            game.undo();
-            
-            if (value >= best) {
-                best = value;
-                bestMove = moves[i];
-            }
-        }
-
-        game.move(bestMove);
-        setFen(game.fen());
-        setGame(game);
-    };
-
-    const minimax = (depth, alpha, beta, wantMax) => {
-        if (depth === 0) return -evalBoard(game);
-
-        var moves = game.moves();
-        let best = 9999 * wantMax ? -1 : 1;
-
-        for (let i = 0; i < moves.length; ++i) {
-            game.move(moves[i]);
-            const bestChild = minimax(depth - 1, alpha, beta, !wantMax);
-            game.undo()
-
-            if (wantMax) {
-                best = Math.max(best, bestChild);
-                alpha = Math.max(alpha, best);
+    const onDrop = ({ sourceSquare, targetSquare }) => {
+      // Attempts to make the player's move.
+      const move = game.move({ from: sourceSquare, to: targetSquare });
+      if (!move) return;
+
+      return new Promise(resolve => {
+          setFen(game.fen());
+          setGame(game);
+          resolve();
+        }).then(() => {
+          setTimeout(() => {
+            if (game.game_over()) {
```

Comment:
you should check for draws as well

Reply:
I think a gameover at that point would actually be a win, you should test it out

Reply:
> [game_over](https://github.com/jhlywa/chess.js#game_over) returns true if the game has ended via checkmate, stalemate, draw, threefold repetition, or insufficient material. Otherwise, returns false.